src/terminal_box.rs

```diff
@@ -587,6 +588,46 @@ where
             state.scrollbar_rect.set(Rectangle::default())
         }
 
+        // Draw cursor
+        {
+            let cursor = terminal.term.lock().renderable_content().cursor;
+            let col = cursor.point.column.0;
+            let line = cursor.point.line.0;
+            let color = terminal.term.lock().colors()[NamedColor::Cursor]
+                .or(terminal.colors()[NamedColor::Cursor])
+                .map(|rgb| Color::from_rgb8(rgb.r, rgb.g, rgb.b))
+                .unwrap_or(Color::WHITE); // TODO default color from theme?
+            let width = terminal.size().cell_width;
+            let height = terminal.size().cell_height;
+            let top_left = view_position
+                + Vector::new((col as f32 * width).floor(), (line as f32 * height).floor());
+            match cursor.shape {
+                CursorShape::Beam => {
+                    let quad = Quad {
+                        bounds: Rectangle::new(top_left, Size::new(1.0, height)),
+                        ..Default::default()
+                    };
+                    renderer.fill_quad(quad, color);
+                }
+                CursorShape::Underline => {
+                    let quad = Quad {
+                        bounds: Rectangle::new(
+                            view_position
+                                + Vector::new(
+                                    (col as f32 * width).floor(),
+                                    ((line + 1) as f32 * height).floor(),
+                                ),
+                            Size::new(width, 1.0),
+                        ),
+                        ..Default::default()
+                    };
+                    renderer.fill_quad(quad, color);
+                }
+                CursorShape::HollowBlock => {} // TODO not sure when this would even be activated
```

Comment:
It's not one of the options in `DECSCUSR` and I couldn't find much info about it...
https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h4-Functions-using-CSI-_-ordered-by-the-final-character-lparen-s-rparen:CSI-Ps-SP-q.1D81

Reply:
```rs
pub enum CursorShape {
    /// Cursor is a block like `▒`.
    #[default]
    Block,

    /// Cursor is an underscore like `_`.
    Underline,

    /// Cursor is a vertical bar `⎸`.
    Beam,

    /// Cursor is a box like `☐`.
    HollowBlock,

    /// Invisible cursor.
    Hidden,
}
```

Reply:
Sure, what I'm not sure about is in what case HollowBlock would be selected.

Reply:
It the application overrides the default cursor.

Reply:
@A-Walrus 

You are write to note that no sequence sets this shape.

It's used by alacritty (the app) for unfocused terminals from the looks of it:
https://github.com/alacritty/alacritty/blob/5060f8eeb864e8c304fbad9588bdd882db942356/alacritty/src/display/content.rs#L57-L58

Reply:
I guess eventually there will a config option for setting the cursor shape, though I am not sure whether the UX team likes hollow cursor.

Reply:
I think it is fine to have HollowBlock not handled yet. The ability to handle underline and beam cursors is already an improvement so I will merge.